spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java

```diff
@@ -465,6 +485,56 @@ private void declareExchange(final String rootName, final Exchange exchange) {
 		addToAutoDeclareContext(rootName + ".exchange", exchange);
 	}
 
+	private void declareConsumerBindings(String name, ExtendedConsumerProperties<RabbitConsumerProperties> properties,
+			Exchange exchange, boolean partitioned, Queue queue) {
+		if (partitioned) {
+			String bindingKey = String.format("%s-%d", name, properties.getInstanceIndex());
+			partitionedBinding(exchange, queue, bindingKey);
+		}
+		else {
+			notPartitionedBinding(exchange, queue, properties.getExtension());
+		}
+	}
+
+	private void partitionedBinding(Exchange exchange, Queue queue, String bindingKey) {
+		if (exchange instanceof TopicExchange) {
+			declareBinding(queue.getName(), BindingBuilder.bind(queue)
+								.to((TopicExchange) exchange)
+								.with(bindingKey));
+		}
+		else if (exchange instanceof DirectExchange) {
+			declareBinding(queue.getName(), BindingBuilder.bind(queue)
+					.to((DirectExchange) exchange)
+					.with(bindingKey));
+		}
+		else if (exchange instanceof FanoutExchange) {
+			throw new IllegalStateException("A fanout exchange is not appropriate for partitioned apps");
+		}
+		else {
+			throw new IllegalStateException("Cannot bind to a " + exchange.getType() + " exchange");
+		}
+	}
+
+	private void notPartitionedBinding(Exchange exchange, Queue queue, RabbitCommonProperties extendedProperties) {
+		if (exchange instanceof TopicExchange) {
+			declareBinding(queue.getName(), BindingBuilder.bind(queue)
+								.to((TopicExchange) exchange)
+								.with("#"));
```

Comment:
>  Only applies to non-partitioned destinations.

This `routingKey` doesn't reflect description.
Something has missed: or here, or in the doc.